helix-rest/src/main/java/org/apache/helix/rest/common/HelixDataAccessorWrapper.java

```diff
@@ -91,6 +93,35 @@ public HelixDataAccessorWrapper(ZKHelixDataAccessor dataAccessor,
     super(dataAccessor);
     _restClient = customRestClient;
     _namespace = namespace;
+    _restSnapShot = new RestSnapshotWrapper(_clusterName);
+  }
+
+  static class RestSnapshotWrapper extends RestSnapShot {
```

Comment:
Why we need this wrapper extend it? If we would like to make this RestSnapShot interface. We can have an impl for it. But if we already have a generic kv snapshot. We dont need to extend the class, right?

Reply:
TFTR. We need to access the generic getter and setter in DataAccessorWrapper, but the getter and setter is private in the RestSnapShot class. So I extends RestSnapShot tp have access to these private functions.